### ISSUE TITLE
Performance: Investigate cocoapods 1.12 concurrency

### ISSUE THREAD

**Issue by drewvolz** [StoDevX/AAO-React-Native#6994]
We are on 1.12 which now can parallelize downloads, speeding up pod installation.

> Using the opt-in :parallel_pod_downloads option, CocoaPods now downloads all pods in parallel, and then moves on to serially install each pod. By default, this is done with 40 threads, which is the sweet spot I found in my testing. However, the number of threads is configurable via the :parallel_pod_download_thread_pool_size option.
> 
> Using 40 threads to parallelize downloads instead of no parallelization results in a 70% speedup for clean pod install times, saving 5 min 17 sec for this particular project.

https://github.com/CocoaPods/CocoaPods/pull/11232

```diff
install! 'cocoapods',
         :deterministic_uuids => false
+        :parallel_pod_downloads => true,
+        :parallel_pod_download_thread_pool_size => 40,
```

**Comment by drewvolz** [comment]
Looking at how things are linked and how this could work.

Our own Podfile calls react native's prepare function

https://github.com/StoDevX/AAO-React-Native/blob/b72b5558195c5687058749d0d8e5af87a6e494c7/ios/Podfile#L5

The prepare function already invokes ` install! 'cocoapods'` located in 
[packages/react-native/scripts/react_native_pods.rb](https://github.com/facebook/react-native/blob/1d0be6d2c8aa71cceef0072513fd8c13eb1aaf15/packages/react-native/scripts/react_native_pods.rb#LL44C1-L52C4)

While we could reimplement this in our own project, we'd have to keep track of what changes upstream? I'm not sure I have a clean way to swizzle what's being called here, short of a patch-package, which I'd rather not resort to.

```diff
diff --git node_modules/react-native/scripts/react_native_pods.rb node_modules/react-native/scripts/react_native_pods.rb
index 5198f80..1dee3f3 100644
--- node_modules/react-native/scripts/react_native_pods.rb
+++ node_modules/react-native/scripts/react_native_pods.rb
@@ -35,7 +35,10 @@ end
 def prepare_react_native_project!
   # Temporary solution to suppress duplicated GUID error.
   # Can be removed once we move to generate files outside pod install.
-  install! 'cocoapods', :deterministic_uuids => false
+  install! 'cocoapods',
+         :deterministic_uuids => false,
+         :parallel_pod_downloads => true,
+         :parallel_pod_download_thread_pool_size => 40
 
   ReactNativePodsUtils.create_xcode_env_if_missing
 end
```

**Comment by drewvolz** [comment]
Tracking upstream [at this issue](https://github.com/facebook/react-native/issues/37318).

**Comment by drewvolz** [comment]
Closed by #7086
